node/src/ledger/read/ledger_read_reducer.rs

```diff
@@ -1,20 +1,419 @@
+                    dispatcher.push(P2pChannelsRpcAction::ResponseSend {
+                        peer_id,
+                        id,
+                        response: resp.as_ref().map(|accounts| {
+                            Box::new(P2pRpcResponse::LedgerQuery(
+                                v2::MinaLedgerSyncLedgerAnswerStableV2::ContentsAre(
+                                    accounts.iter().cloned().collect(),
+                                ),
+                            ))
+                        }),
+                    });
+                }
+            }
+            (req, LedgerReadResponse::GetStagedLedgerAuxAndPendingCoinbases(resp)) => {
+                for (peer_id, id, is_streaming) in find_peers_with_ledger_rpc(state, req) {
+                    if is_streaming {
+                        dispatcher.push(P2pChannelsStreamingRpcAction::ResponseSendInit {
+                            peer_id,
+                            id,
+                            response: resp.clone().map(Into::into),
+                        });
+                    } else {
+                        dispatcher.push(P2pChannelsRpcAction::ResponseSend {
+                            peer_id,
+                            id,
+                            response: resp.clone().map(|data| {
+                                Box::new(P2pRpcResponse::StagedLedgerAuxAndPendingCoinbasesAtBlock(
+                                    data,
+                                ))
+                            }),
+                        });
+                    }
+                }
+            }
+            (
+                LedgerReadRequest::ScanStateSummary(ledger_hash),
+                LedgerReadResponse::ScanStateSummary(scan_state),
+            ) => {
+                for rpc_id in state
+                    .rpc
+                    .scan_state_summary_rpc_ids()
+                    .filter(|(_, hash, _)| *hash == ledger_hash)
+                    .map(|(id, ..)| id)
+                    .collect::<Vec<_>>()
+                {
+                    dispatcher.push(RpcAction::ScanStateSummaryGetSuccess {
+                        rpc_id,
+                        scan_state: scan_state.clone(),
+                    });
+                }
+            }
+            (_, LedgerReadResponse::ScanStateSummary(..)) => unreachable!(),
+            (_req, LedgerReadResponse::GetAccounts(..)) => todo!(),
+            (_, LedgerReadResponse::AccountsForRpc(rpc_id, accounts, account_query)) => {
+                dispatcher.push(RpcAction::LedgerAccountsGetSuccess {
+                    rpc_id,
+                    accounts,
+                    account_query,
+                });
+            }
+        }
+    }
+
+    fn next_read_requests_init(dispatcher: &mut Dispatcher<Action, State>, state: &State) {
+        // fetching delegator table
+        dispatcher.push(BlockProducerVrfEvaluatorAction::BeginDelegatorTableConstruction);
```

Comment:
hmmm I am wondering why this is even here. @binier do you remember?

Reply:
@tizoc Because delegator table construction requires reading from ledger. It could be that ledger read quota was reached when vrf tried to initiate that read, so we need to "retry" it if that's the case.

Reply:
Ok, can you add a comment mentioning that @0xMimir ?